301-azure-firewall-dns-proxy/azuredeploy.json

```diff
@@ -0,0 +1,2743 @@
+    "blobPrivateDnsZoneName": "[concat('privatelink', variables('blobPublicDNSZoneForwarder'))]",
+    "adlsPrivateDnsZoneId": "[resourceId('Microsoft.Network/privateDnsZones', variables('adlsPrivateDnsZoneName'))]",
+    "blobPrivateDnsZoneId": "[resourceId('Microsoft.Network/privateDnsZones', variables('blobPrivateDnsZoneName'))]",
+    "adlsServicePrimaryEndpoint": "[concat(parameters('adlsStorageAccountName'), variables('adlsPublicDNSZoneForwarder'))]",
+    "blobServicePrimaryEndpoint": "[concat(parameters('blobStorageAccountName'), variables('blobPublicDNSZoneForwarder'))]",
+    "adlsStorageAccountAdlsPrivateEndpointId": "[resourceId('Microsoft.Network/privateEndpoints', parameters('adlsStorageAccountAdlsPrivateEndpointName'))]",
+    "adlsStorageAccountBlobPrivateEndpointId": "[resourceId('Microsoft.Network/privateEndpoints', parameters('adlsStorageAccountBlobPrivateEndpointName'))]",
+    "blobStorageAccountBlobPrivateEndpointId": "[resourceId('Microsoft.Network/privateEndpoints', parameters('blobStorageAccountBlobPrivateEndpointName'))]",
+    "adlsGroupName": "dfs",
+    "blobGroupName": "blob",
+
+    "dnsVmScriptFileUri": "[uri(parameters('_artifactsLocation'), concat('scripts/', parameters('dnsVmScriptFileName'), parameters('_artifactsLocationSasToken')))]",
+    "testVmScriptFileUri": "[uri(parameters('_artifactsLocation'), concat('scripts/', parameters('testVmScriptFileName'), parameters('_artifactsLocationSasToken')))]"
+
+  },
+  "resources": [
+    {
+      "apiVersion": "2019-06-01",
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('adlsStorageAccountName')]",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "Standard_RAGRS"
+      },
+      "kind": "StorageV2",
+      "properties": {
+        "networkAcls": {
+          "defaultAction": "[parameters('adlsStorageAccountNetworkAclsDefaultAction')]",
+          "bypass": "AzureServices"
+        },
+        "allowBlobPublicAccess": "[parameters('adlsStorageAccountAllowBlobPublicAccess')]"
+      }
+    },
+    {
+      "apiVersion": "2019-06-01",
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('blobStorageAccountName')]",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "kind": "StorageV2",
+      "properties": {
+        "networkAcls": {
+          "defaultAction": "[parameters('blobStorageAccountNetworkAclsDefaultAction')]",
+          "bypass": "AzureServices"
+        },
+        "allowBlobPublicAccess": "[parameters('blobStorageAccountAllowBlobPublicAccess')]"
+      }
+    },
+    {
+      "comments": "Network Security Group for the Subnets of the Hub Virtual Network",
+      "type": "Microsoft.Network/networkSecurityGroups",
+      "apiVersion": "2020-07-01",
+      "name": "[variables('hubVnetCommonSubnetNsgName')]",
+      "location": "[parameters('location')]",
+      "resources": [
+        {
```

Comment:
let's refactor the diags resources to use the scope property - easier to author and will translate better to bicep

Reply:
I didn't get your request Brian, can we just skip this? I think the syntax is correct, Bicep will wait 😄